### DISCUSSION TITLE
Installing pympress without install latest python version

### DISCUSSION THREAD

**Discussion by anandkumar89** [Cimbali/pympress#289] [Q&A] [answered]
configuration : Intel Mac + python3.9.14

I want to install pympress but this would install python3.11. I do not want multiple python version on my system. Is there force installation using libraries compatible with installed python version ?

Thank you!

**Comment by Cimbali** [comment, marked as answer]
If I understand well, installing through homebrew would cause a python3.11 install you’d want to avoid?

I think it should be doable though not necessarily easy. You‘ll need to install all [dependencies](https://github.com/Cimbali/pympress#dependencies) either in brew or outside (mainly [Gtk](https://www.gtk.org/docs/installations/macos/) and Poppler). Then you should install the python dependencies with pip of your current python.

The complex bits are going to be:
- installing python3-gobject (which requires some building to be done, so build tools + dev headers on the dependencies)
- at runtime, getting your python gobject library to find the typelib files, that contain the introspection information of the dependency libraries (Gtk, cairo, Poppler, and the like) and are necessary to make python interact with these libraries

If you succeed, please come back to leave indications on how you solved these issues for future users :)
